src/main/java/de/tum/in/www1/exerciseapp/web/rest/ExerciseResource.java

```diff
@@ -148,7 +144,7 @@ public ExerciseResource(ExerciseRepository exerciseRepository, ExerciseService e
      * @return the ResponseEntity with status 200 (OK) and the list of exercises in body
      */
     @GetMapping("/exercises")
-    @PreAuthorize("hasAnyRole('ADMIN', 'TA')")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     @Timed
     public ResponseEntity<List<Exercise>> getAllExercises(@ApiParam Pageable pageable) {
```

Comment:
This should only be possible for TAs and instrcutors of this particular course (and admins)

Reply:
I wasn't able to identify where this method is used. According to my observations, GET /course/:courseId/exercises is used to retrieve exercises for a particular course, and this is done for all courses. Do you know the usage of this method?

Reply:
Done.